net-dns/dnscrypt-proxy/dnscrypt-proxy-2.0.0.ebuild

```diff
@@ -0,0 +1,80 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+EGO_PN="github.com/jedisct1/dnscrypt-proxy/..."
+EGO_SRC="github.com/jedisct1/dnscrypt-proxy"
+
+inherit systemd user golang-build
+
+DESCRIPTION="A flexible DNS proxy, with support for modern encrypted DNS protocols."
+HOMEPAGE="https://${EGO_PN}"
+SRC_URI="https://${EGO_PN}/archive/2.0.0.tar.gz -> ${P}.tar.gz"
+
+LICENSE="ISC"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~x86"
+IUSE="systemd"
+
+RDEPEND="systemd? ( sys-apps/systemd )"
+
+S="${WORKDIR}/${P}"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-2.0.0_fix_systemd_service.patch"
+)
+
+pkg_setup() {
+	enewgroup dnscrypt
+	enewuser dnscrypt -1 -1 /var/empty dnscrypt
+}
```

Comment:
Why create a user/group if it's not even used?
And even if you run it as user it'll fail to bind to port 53.
Need to set capabilities cap_net_bind_service=+pe for it to be able to bind privileged port as non-root.

Reply:
We can add "DynamicUser=yes" and 
"CapabilityBoundingSet=CAP_NET_BIND_SERVICE" and "AmbientCapabilities=CAP_NET_BIND_SERVICE " to service.
For openrc we still need user and setcap call.

Reply:
Well, we have to support both world (openrc and systemd).
And since openrc is the default and official we have to create user anyway. Why not just reuse it?
Systemd's capabilities seems like a nice measure to ensure it can bind to port 53 if fcaps ever fails.

I'll remove sed lines in my version and just ship custom service file based on upstream's.